<launch>

    <arg name="use_cable_sensors" default="true"/>
    <!-- <arg name="frame" default="data"/> -->
    <arg name="cable_unit_conversion"/>
    <arg name="cable_map_filename"/>
	<arg name="initial_estimate_filename"/>
	
	<node pkg="estimation" type="ekf_estimation" name="ekf_estimation" output="screen">
        <param name="red_endcaps_topic" value="/tensegrity/endcap/red/positions"/>
        <param name="blue_endcaps_topic" value="/tensegrity/endcap/blue/positions"/>
        <param name="green_endcaps_topic" value="/tensegrity/endcap/green/positions"/>
        <param name="tensegrity_pose_topic" value="/tensegrity/poses"/>
        <!-- <param name="cables_topic" value="/tensegrity/cables"/> -->
        <param name="sensors_topicname" value="/tensegrity/sensors"/>
        <param name="use_cable_sensors" value="$(arg use_cable_sensors)"/>
        <param name="frequency" value="30"/>
        <param name="cable_unit_conversion" value="$(arg cable_unit_conversion)"/>
        <param name="cable_map_filename" value="$(arg cable_map_filename)"/>
        <param name="initial_estimate_filename" value="$(arg initial_estimate_filename)"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="viz_ekf_nodelet"  args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="BarsToMarkersStateEstimation" 
        args="load interface/TensegrityBarsToMarkers viz_ekf_nodelet" output="screen">
        <param name="tensegrity_bars_topicname" value="/tensegrity/poses"/>
        <param name="tensegrity_markers_topicname" value="/tensegrity/markers"/>
        <param name="node_id" value="/nodes/BarsToMarkersStateEstimation"/>
    </node>
</launch>